pkg/workerpool/workerpool.go

```diff
@@ -37,9 +37,17 @@ type WorkerPool struct {
 
 // Job is a runnable interface to queue jobs on a WorkerPool
 type Job interface {
-	JobName() string // Returns name for a job
-	Hash() uint64    // Returns a uint64 hash for a job.
-	Run()            // Executes the job
+	// JobName returns the name of the job.
+	JobName() string
+
+	// Hash returns a uint64 hash for a job.
+	Hash() uint64
+
+	// Run executes the job.
+	Run()
+
+	// GetDoneCh returns the channel, which when closed, indicates that the job was finished.
```

Comment:
I was thinking if there'd be a way to hide this in the api and have instead the workerpool module provide this functionality transparently without having the job have to take care of it...

this seems sort of common code that all jobs will implement, so if we were able to hide, it could be better

Reply:
@eduser25 Are you ok if we merge this and you can tweak this with how you imagine this evolving?  Given that we don't have any other uses for the Job system - we could make changes as we wish.

Reply:
sure, go ahead